File: tags/xmount-0.4.7/libewf/examples/ewf.net/Glob.vb
﻿REM VB.Net program to glob Expert Witness Compression Format file(s) using ewf.net
REM
REM Author:            Joachim Metz
REM Creation date:     November 3, 2010
REM Modification date: July 24, 2011

Imports EWF

Module Glob

Function Main(ByVal args() As String) As Integer
        Dim handle As New EWF.Handle 
        Dim ewf As New EWF.EWF   

	Dim filenames As Array = Nothing
	Dim filename As String = Nothing
	Dim program As String  = "Glob.exe"

	Console.WriteLine(program + " (ewf.net " + ewf.GetVersion() + ")")

	If args.Length() <> 1 Then
		Console.WriteLine("Usage: " + program + " filename")

		Return 1
	End If

	filenames = handle.Glob(args(0))

	If filenames.Length() > 0 Then
		Console.WriteLine("Filenames:")

		For Each filename In filenames
			Console.WriteLine(filename)
		Next
	End If

	Console.WriteLine("")

	Return 0
End Function

End Module

File: tags/xmount-0.5.0/libewf/examples/pyewf/glob.py
#! /usr/bin/env python
#
# Python script to glob Expert Witness Compression format file(s) using pyewf
#
# Author:            Joachim Metz
# Creation date:     October 14, 2010
# Modification date: January 4, 2011
#

__author__    = "Joachim Metz"
__version__   = "20110104"
__date__      = "Jan 4, 2011"
__copyright__ = "Copyright (c) 2006-2012, Joachim Metz <jbmetz@users.sourceforge.net>"
__license__   = "GNU LGPL version 3"

import sys
import pyewf

# ----------------------------------------------------------------------------
# Main
# ----------------------------------------------------------------------------

print "glob.py " + __version__ + " (libewf " + pyewf.get_version() + ")\n"

argc = len( sys.argv )

if argc != 2:
	print "Usage: glob.py filename\n"

	sys.exit( 1 )

try:
	filenames = pyewf.glob(
	             sys.argv[ 1 ] )

except:
	print "Unable to glob filename(s)\n"
	print sys.exc_info()[ 1 ]

	sys.exit( 1 )

if len( filenames ) > 0:
	print "Filenames:"

	for filename in filenames:
		print filename.encode( "utf8" )

	print ""

sys.exit( 0 )

